streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -1250,6 +1253,20 @@ private Optional<String> removeStreamThread(final long timeoutMs) throws Timeout
         return Optional.empty();
     }
 
+    private int calculateMetricsRecordingLevel() {
+        final int recordingLevel;
+        final String recordingLevelString = applicationConfigs.getString(METRICS_RECORDING_LEVEL_CONFIG);
+        if (recordingLevelString.equals("INFO")) {
+            recordingLevel = 0;
+        } else if (recordingLevelString.equals("DEBUG")) {
+            recordingLevel = 1;
+        } else {
+            // Must be TRACE level
```

Comment:
Might be better to use another `if` and final `else` throw an exception as safe guard? Would also highlight that we need to update this code, in case we add a new level?